CitationStyleLanguagePlugin.php

```diff
@@ -0,0 +1,1048 @@
+        }
+        return $success;
+    }
+
+    /**
+     * Get list of citation styles available
+     *
+     * @return array
+     */
+    public function getCitationStyles()
+    {
+        if (!empty($this->_citationStyles)) {
+            return $this->_citationStyles;
+        }
+
+        $defaults = [
+            [
+                'id' => 'acm-sig-proceedings',
+                'title' => __('plugins.generic.citationStyleLanguage.style.acm-sig-proceedings'),
+                'isEnabled' => true,
+            ],
+            [
+                'id' => 'acs-nano',
+                'title' => __('plugins.generic.citationStyleLanguage.style.acs-nano'),
+                'isEnabled' => true,
+            ],
+            [
+                'id' => 'apa',
+                'title' => __('plugins.generic.citationStyleLanguage.style.apa'),
+                'isEnabled' => true,
+                'isPrimary' => true,
+            ],
+            [
+                'id' => 'associacao-brasileira-de-normas-tecnicas',
+                'title' => __('plugins.generic.citationStyleLanguage.style.associacao-brasileira-de-normas-tecnicas'),
+                'isEnabled' => true,
+            ],
+            [
+                'id' => 'chicago-author-date',
+                'title' => __('plugins.generic.citationStyleLanguage.style.chicago-author-date'),
+                'isEnabled' => true,
+            ],
+            [
+                'id' => 'harvard-cite-them-right',
+                'title' => __('plugins.generic.citationStyleLanguage.style.harvard-cite-them-right'),
+                'isEnabled' => true,
+            ],
+            [
+                'id' => 'ieee',
+                'title' => __('plugins.generic.citationStyleLanguage.style.ieee'),
+                'isEnabled' => true,
+            ],
+            [
+                'id' => 'modern-language-association',
+                'title' => __('plugins.generic.citationStyleLanguage.style.modern-language-association'),
+                'isEnabled' => true,
+            ],
+            [
+                'id' => 'turabian-fullnote-bibliography',
+                'title' => __('plugins.generic.citationStyleLanguage.style.turabian-fullnote-bibliography'),
+                'isEnabled' => true,
+            ],
+            [
+                'id' => 'vancouver',
+                'title' => __('plugins.generic.citationStyleLanguage.style.vancouver'),
+                'isEnabled' => true,
+            ],
+        ];
+
+        // If hooking in to add a custom .csl file, add a `useCsl` key to your
+        // style definition with the path to the file.
+        Hook::call('CitationStyleLanguage::citationStyleDefaults', [&$defaults, $this]);
+        $this->_citationStyles = $defaults;
+
+        return $this->_citationStyles;
+    }
+
+    /**
+     * Get the primary style name or default to the first available style
+     *
+     * @param $contextId integer Journal ID
```

Comment:
This applies to this line and a few others below:

We're relying less on phpdoc-style self-documentation and using more typehints recently. Since this is an application-independent repository, we shouldn't be referring to OJS-specific terminology, so this comment should read "Context ID" -- but that adds nothing to the variable name :) so I'd suggest getting rid of the `@param` line entirely and adding a typehint to the function parameter.

Self-documentation for parameters and returns is fine when it adds value, but often it doesn't.

I don't think it's possible for this code to get called when `$contextId` is `0`, so I'd suggest removing the default value entirely and also up the check at https://github.com/pkp/citationStyleLanguage/pull/84/files#diff-13b0240e88db06d12f97fde1af2328104d06b03bdab4c6b2b5e7bfa949e9be1aR350, assuming instead that `$context` will always exist as it should. I'd rather have a fatal error in a supposedly-unrunnable case than have it magically transform an ID into a `0`; we're getting stricter about this in the codebase now that we're e.g. using referential integrity in the database.

(It is possible for article summaries to be displayed where `$contextId` is `0` -- e.g. in the site-wide search results page. But that's just the article summary, not the article view.)

(Sorry -- this is getting long! -- but if there *was* a valid case with `$contextId = 0`, then I would suggest instead using the constant `CONTEXT_SITE` from `PKPApplication.php`.)